src/jvm/main/org/jetbrains/kotlinx/lincheck/transformation/LincheckClassVisitor.kt

```diff
@@ -121,13 +122,159 @@ internal class LincheckClassVisitor(
             sv.analyzer = aa
             aa
         }
+        mv = CoverageBytecodeFilter(
+            skipVisitor.newAdapter(),
+            mv.newAdapter()
+        )
         mv = DeterministicHashCodeTransformer(fileName, className, methodName, mv.newAdapter())
         mv = DeterministicTimeTransformer(mv.newAdapter())
         mv = DeterministicRandomTransformer(fileName, className, methodName, mv.newAdapter())
         return mv
     }
 
 }
+    /**
+     * Filters out chunks of bytecode generated by coverage.
+     *
+     * @param initialAdapter initial adapter to delegate work to when coverage-related bytecode encountered.
+     * @param nextAdapter adapter to delegate calls to when bytecode is not related to coverage.
+     */
+    private class CoverageBytecodeFilter(
```

Comment:
Please move `CoverageBytecodeFilter` into a separate file in the `transformers` sub-package, similarly as it is done for other byte-code transformers.